src/objectExplorer/objectExplorerService.ts

```diff
@@ -135,6 +135,7 @@ export class ObjectExplorerService {
                 const credentials = self._sessionIdToConnectionCredentialsMap.get(result.sessionId);
                 const children = result.nodes.map(node => TreeNodeInfo.fromNodeInfo(node, result.sessionId,
                     self._currentNode, credentials));
+                self._currentNode = self._rootTreeNodeArray.find(n => n.nodePath === result.nodePath);
```

Comment:
This isn't really going to solve the inherent problem that you have a huge potential for race conditions here. There's other callbacks that use self._currentNode so depending on the order the events are handled that could result in a lot of subtle bugs. 

In the end I'm not even sure what "currentNode" in this context would mean. The last one to react to an event? The currently selected one in the tree? Something else? I feel like the definition is pretty vague which is why you're having problems like this so would suggest looking at fixing that instead of doing one-off fixes like this (that might result in other issues popping up if things aren't expecting currentNode to be changed by this handler)